packages/next/src/server/image-optimizer.ts

```diff
@@ -141,6 +142,11 @@ export function detectContentType(buffer: Buffer) {
   ) {
     return AVIF
   }
+
+  if (buffer.length >= 4 && buffer[0] === 0x00 && buffer[1] === 0x00) {
+    return ICO
+  }
```

Comment:
Can you share the documentation where you found this algorithm?

It seems too simple. The wiki page seems to indicate the value `1` is used to detect `.ico`

<img width="760" alt="image" src="https://user-images.githubusercontent.com/229881/224403831-d96e27e5-81e3-4db0-815c-b3e985af5427.png">

Reply:
@styfle Thank you for response! I will be honest and I can say that I wrote this line using Github Copilot :) After this, I wrote the test and made sure that everything works.

If you believe that this solution can create problems - I can investigate this topic and change it.

Reply:
I think we can change it to:
```js
if ([0x00, 0x00, 0x01, 0x00].every((b, i) => buffer[i] === b)) {
    return ICO
  }
```

Reply:
@styfle I've updated the PR applying your suggestion 🙏